server/src/QRServer/db/connector.py

```diff
@@ -197,6 +195,36 @@ async def get_match(self, match_id: str) -> Optional[DbMatchReport]:
             is_void=row[11],
         )
 
+    async def get_match2(self, match_id: str):
+        c = await self.conn.cursor()
+
+        await c.execute(
+            "select"
+            " u1.username,"
+            " u2.username,"
+            " m.winner_pieces_left,"
+            " m.loser_pieces_left,"
+            " m.started_at,"
+            " m.finished_at,"
+            " m.move_counter"
+            " from matches m"
+            " left join users u1 on m.winner_id = u1.id"
+            " left join users u2 on m.loser_id = u2.id"
+            " where m.id = ?", (match_id,))
+
+        row = await c.fetchone()
+        if row is None:
+            return None
+        return GameResultHistory(
```

Comment:
I would keep this as something akin to Response message and instead make a new model

Reply:
The new model could return this class from match record